packages/bot-web-ui/src/pages/server-side-bot/add-bot/form.tsx

```diff
@@ -0,0 +1,200 @@
+                    {group.map((field, field_index) => {
+                        const key = `${field.name || field.type} + ${field_index}`;
+
+                        if (
+                            (!is_desktop && field.hide?.includes('mobile')) ||
+                            (is_desktop && field.hide?.includes('desktop'))
+                        ) {
+                            return null;
+                        }
+
+                        switch (field.type) {
+                            // Generic or common fields
+                            case 'text': {
+                                return (
+                                    <QSInput
+                                        {...field}
+                                        key={key}
+                                        onChange={onChange}
+                                        name={field.name as string}
+                                        has_currency_unit={false}
+                                    />
+                                );
+                            }
+                            case 'number': {
+                                if (!field.name) return null;
+                                const {
+                                    should_have = [],
+                                    hide_without_should_have = false,
+                                    has_currency_unit = false,
+                                } = field;
+                                const should_enable = shouldEnable(should_have);
+                                const initial_stake = 1;
+                                let min = 1;
+                                let max;
+                                if (field.name === 'duration' && current_duration_min_max) {
+                                    min = current_duration_min_max.min;
+                                    max = current_duration_min_max.max;
+                                }
+                                const should_validate = field.should_have;
+                                if (should_validate && field.name === 'max_stake') {
+                                    min = +form_data?.stake;
+                                    if (isNaN(min)) {
+                                        min = +initial_stake;
+                                    }
+                                }
+                                if (should_validate && field.name === 'last_digit_prediction') {
+                                    if (
+                                        isNaN(+form_data?.last_digit_prediction) ||
+                                        +form_data?.last_digit_prediction === 1
+                                    ) {
+                                        min = 0;
+                                    }
+                                    if (+form_data?.last_digit_prediction > 0) {
+                                        min = +form_data?.last_digit_prediction - 1;
```

Comment:
Can we extract all this logic of determining what's `min` and what's `max` from each of the `case`s into an external helper?  🙏🏻
it would be help reduce the boilerplate in this file.

Reply:
Will be done in the refactoring phase post release 
thanks @maryia-deriv 🙇